bluetooth_mesh/mesh.py

```diff
@@ -347,3 +352,87 @@ def unpack(cls, app_key: ApplicationKey, net_key: NetworkKey, local_iv_index: in
             transport_msg = AccessMessage.decrypt(app_key, iv_index, ctl, ttl, seq, src, dst, transport_pdu)
         net_message = NetworkMessage(transport_msg)
         return iv_index, seq, net_message
+
+
+mesh_crc = Configuration(
+        width=8,
+        polynomial=0x07,
+        init_value=0xff,
+        final_xor_value=0xff,
+        reverse_input=True,
+        reverse_output=True
+    )
+
+
+class GenericProvisioningPDU:
+
+    @staticmethod
+    def pack(payload):
+        if isinstance(payload['type'], ProvisioningPDUType):
+            PDU = ProvisioningPDU.build(obj=payload)
+            segments = [PDU[0:20]] + ([PDU[0+i:23+i] for i in range(20, len(PDU), 23)] if len(PDU) > 20 else [])
+            total_len = len(PDU)
+            fcs = CrcCalculator(configuration=mesh_crc).calculate_checksum(PDU)
+
+            ret = list()
+            ret.append(
+                TransactionStartPDU.build(
+                    dict(
+                        last_segment_number=len(segments)-1,
+                        total_length=total_len,
+                        frame_check=fcs,
+                        data=segments.pop(0)
+                    )
+                )
+            )
+
+            for index, segment in enumerate(segments, start=1):
+                ret.append(
+                    TransactionContinuationPDU.build(
+                        dict(
+                            segment_index=index,
+                            data=segment
+                        )
+                    )
+                )
+
+            return ret
+
+        elif isinstance(payload['type'], BearerOpcode):
```

Comment:
make this an `if`, move at the beginning and remove `elif` (`else` is redundant after `return`)